### ISSUE TITLE
[java] StringToString doesn't trigger on string literals

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3681]
**Affects PMD Version:6.41.0** 

**Rule:StringToString**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#stringtostring>

**Description:**
I found a false negative about the rule StringToString, please refer to the minimized sample below.

PMD should have reported a warning at line 3, but it does not. However, this is definitely a String object calls `toString()` method. So, I think this is a false negative.

**Code Sample demonstrating the issue:**
```java
class Impl {
  String foo() {
    return "123".toString();
  }
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
